src/AutoMapper/MapperConfiguration.cs

```diff
@@ -244,6 +254,25 @@ public bool AllowNullCollections
                     //});
                 }
 
+                var genericMapperType = FindClosedGenericObjectMapperFor(types);
+
+                if (genericMapperType != null)
+                {
+                    var srcParam = Parameter(typeof(TSource), "src");
+                    var destParam = Parameter(typeof(TDestination), "dest");
+                    var ctxtParam = Parameter(typeof(ResolutionContext), "ctxt");
+
+                    var mapExpression = Lambda(
+                        ToType(Call(
+                            New(genericMapperType),

```

Comment:
I still hate you have to new it up every time.  I would much rather it be static call.  Yes you lose interfaces, but at least save on newing up an object on every map.